hq_superset/api.py

```diff
@@ -67,3 +68,8 @@ def post_dataset_change(self) -> FlaskResponse:
                 'Invalid JSON syntax',
                 status=HTTPStatus.BAD_REQUEST.value,
             )
+        except TableMissing:
+            return json_error_response(
+                'Data source not found',
+                status=HTTPStatus.HTTP_404_NOT_FOUND.value,
```

Comment:
will this cause the repeater to constantly keep trying? in that case this won't be better than what this currently is.

Reply:
It won't stop the repeater from retrying, but I thought this change at least gives more info w.r.t why the request is failing. Ideally this won't happen after we properly handle data source table removals / repeater removals, but having this feels a little better than is.

That said, I'm willing to let this go, but let's hear what @kaapstorm says.

Reply:
On second thought, I think this is still good to have irrespective of how it is being used in external service as a good REST practice.

Reply:
Yeah, I second @ajeety4 's second thought. If a sender (even an API user) expects a thing to be there, and it isn't there, a 404 response sounds right to me.

What's more, the alternative is a 500, which suggests that the problem lies with the server, not the sender, and that sounds wrong. The sender should be told that they are trying to update a DataSet that isn't there.

Reply:
Okay :+1: